nix-installer.nix

```diff
@@ -0,0 +1,172 @@
+{ installationSlug ? "nix-install", installedPackages, postInstall ? null }:
+let
+  pkgs = import (import ./fetchNixpkgs.nix (builtins.fromJSON (builtins.readFile ./nixpkgs-src.json))) { config = {}; overlays = []; };
+  nix-bundle = import (pkgs.fetchFromGitHub {
+    owner = "input-output-hk";
+    repo = "nix-bundle";
+    rev = "630e89d1d16083";
+    sha256 = "1s9vzlsfxd2ym8jzv2p64j6jlwr9cmir45mb12yzzjr4dc91xk8x";
+  }) { nixpkgs = pkgs; };
+  installerBundle = nix-bundle.nix-bootstrap {
+    target = "${installer}";
+    run = "/bin/installer";
+    nixUserChrootFlags = "-c -m /home:/home -p HOME";
+  };
+  nixSrc = pkgs.fetchFromGitHub {
+    owner = "nixos";
+    repo = "nix";
+    rev = "16551f54c94f2b551ebaf00a7bd0245dc3b0b9e4";
+    sha256 = "0kd13v4xl4imwb3141pnn0lqx0xfcmgnwd026c10kmvjhm848pwx";
+  };
+  nixFix = pkgs.nixUnstable.overrideDerivation (drv: { src = nixSrc; });
+  utils = pkgs.writeText "utils.sh" ''
+    function rmrf {
+      chmod -R +w "$*" || true
+      rm -rf "$*" || true
+    }
+  '';
+  updater = pkgs.writeScriptBin "update-runner" ''
+    #!${pkgs.stdenv.shell}
+
+    source ${utils}
+
+    set -xe
+
+    if [ ! -e "$1" ]; then
+      echo "update file not found"
+      exit -1
+    fi
+    pwd
+    id
+    UNPACK=$(mktemp -d)
+    cd $UNPACK
+    echo "$@"
+    "$1" --extract
+    ls -ltrh dat/nix/store/*-tarball/tarball/tarball.tar.xz
+    UNPACK2=$(mktemp -d)
+    tar -C $UNPACK2 -xf dat/nix/store/*-tarball/tarball/tarball.tar.xz
+    cd
+    rmrf $UNPACK
+    ls -ltrh $UNPACK2
+    NIX_REMOTE=local?root=$UNPACK2 nix-store --load-db < $UNPACK2/nix-path-registration
+    NIX_REMOTE=local?root=$UNPACK2 nix-store --verify --check-contents
+    nix copy --no-check-sigs --from local?root=$UNPACK2 $(readlink $UNPACK2/firstGeneration)
+    export NIX_PROFILE=$DIR/nix/var/nix/profiles/profile
+    nix-env --set $(readlink $UNPACK2/firstGeneration)
+    rmrf $UNPACK2
+
+    post-install || true
```

Comment:
Shouldn't this also launcher Daedalus as well? I'd imagine it would start up after installation.